core/ibft.go

```diff
@@ -623,7 +626,7 @@ func (i *IBFT) validateProposal0(msg *proto.Message, view *proto.View) bool {
 	)
 
 	//	proposal must be for round 0
-	if msg.View.Round != 0 {
+	if messages.ExtractProposal(msg).Round != 0 {
```

Comment:
Unrelated to this check here, but wanted to point out an additional check is missing (both from the pseudocode and the implementation), and it is referenced as part of `Issue N`:

There is no check that the `r_pp` from the round change message matches the round from the proposal message (`r_EB`), and this check is required for this to be safe.

Please implement this, along with a coverage test

Reply:
Added the check in [a03cc81](https://github.com/madz-lab/go-ibft/pull/10/commits/a03cc81bb13f369e0e5a4062e9205b3af3f6498a)

The check is already triggered with existing tests effectively bypassing subsequent checks triggered by the other cases (mostly due to how messages are generated with test helpers)

The testing suite has been greatly improved in v2 branch, so I won't be amending the cases present on this one